packages/google_sign_in/test/google_sign_in_test.dart

```diff
@@ -251,4 +253,40 @@ void main() {
       expect(await googleSignIn.signIn(), isNotNull);
     });
   });
+
+  group('GoogleSignIn with fake backend', () {
+    const MethodChannel channel = const MethodChannel(
+      'plugins.flutter.io/google_sign_in',
+    );
+
+    const Map<String, String> kUserData = const <String, String>{
+      "displayName": "John Doe",
+      "email": "john.doe@gmail.com",
+      "id": "8162538176523816253123",
+      "photoUrl": "https://lh5.googleusercontent.com/photo.jpg",
+    };
+
+    GoogleSignIn googleSignIn;
+
+    setUp(() {
+      channel.setMockMethodCallHandler(
+          (new FakeSignInBackend()..setUser(kUserData)).handleMethodCall);
+      googleSignIn = new GoogleSignIn();
+    });
+
+    test('user starts as null', () async {
+      expect(googleSignIn.currentUser, isNull);
+    });
+
+    test('can sign in', () async {
+      await googleSignIn.signIn();
+      expect(googleSignIn.currentUser.toString(),
+          equals('GoogleSignInAccount:$kUserData'));
+    });
+
+    test('can sign out', () async {
+      await googleSignIn.disconnect();
```

Comment:
You should combine this into the "can sign in" test.  Since googleSignIn starts out with no current user, calling `disconnect()` without having first signed in is effectively a no-op.  So having a test that tests "can sign in and then sign out" ensures that you're testing the full flow.

That said, having a distinct test that tests "disconnect when already signed out succeeds" is valid.